src/posit/connect/resources.py

```diff
@@ -61,77 +61,108 @@ def __init__(self, ctx: Context, path: str, pathinfo: str = "", /, **attributes)
-        ----------
-        _ctx : Context
-            The context object containing the session and URL for API interactions
-        _path : str
-            The HTTP path for the collection endpoint.
         """
         params = ResourceParameters(ctx.session, ctx.url)
         super().__init__(params, **attributes)
         self._ctx = ctx
-        self._path = posixpath.join(path, pathinfo)
+        self._path = path
 
 
 T = TypeVar("T", bound="Active")
 """A type variable that is bound to the `Active` class"""
 
 
 class ActiveSequence(ABC, Generic[T], Sequence[T]):
-    def __init__(self, ctx: Context, path: str, pathinfo: str = "", uid: str = "guid"):
-        """A sequence abstraction for any HTTP GET endpoint that returns a collection.
+    """A sequence for any HTTP GET endpoint that returns a collection."""
 
-        It lazily fetches data on demand, caches the results, and allows for standard sequence operations like indexing and slicing.
+    _cache: Optional[List[T]]
 
-        Attributes
+    def __init__(self, ctx: Context, path: str, uid: str = "guid"):
+        """A sequence abstraction for any HTTP GET endpoint that returns a collection.
+
+        Parameters
         ----------
-        _ctx : Context
-            The context object containing the session and URL for API interactions
-        _path : str
-            The HTTP path for the collection endpoint.
-        _uid : str
-            The field name used to uniquely identify records.
-        _cache: Optional[List[T]]
+        ctx : Context
+            The context object containing the session and URL for API interactions.
+        path : str
+            The HTTP path component for the collection endpoint
+        uid : str, optional
+            The field name of that uniquely identifiers an instance of T, by default "guid"
         """
         super().__init__()
         self._ctx = ctx
-        self._path = posixpath.join(path, pathinfo)
+        self._path = path
         self._uid = uid
-        self._cache: Optional[List[T]] = None
+        self._cache = None
+
+    @abstractmethod
+    def _create_instance(self, path: str, /, **kwargs: Any) -> T:
+        """Create an instance of 'T'."""
+        raise NotImplementedError()
+
+    def cached(self) -> bool:
+        """Returns True if the collection is cached.
 
-    def _get_or_fetch(self) -> List[T]:
+        Returns
+        -------
+        bool
+
+        See Also
+        --------
+        reload
         """
-        Fetch and cache the data from the API.
+        return self._cache is not None
 
-        This method sends a GET request to the `_endpoint` and parses the response as a list of JSON objects.
-        Each JSON object is used to instantiate an item of type `T` using the class specified by `_cls`.
-        The results are cached after the first request and reused for subsequent access unless reloaded.
+    def reload(self) -> Self:
+        """Reloads the collection from Connect.
```

Comment:
Mark for internal usage. Match name to actions performed
```suggestion
    def _reset_cache(self) -> Self:
        """Resets the collected values from Connect.
```

Reply:
This one is intentionally public to give the caller control over refreshing the collection from the server. Otherwise, there is not way for them to do so. I think `reload` flows better than `reset`. For example, `jobs.reload().find_by(hostname='example.com')` vs `jobs.reset().find_by(hostname='example.com')`